app/models/content_block.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class ContentBlock < ApplicationRecord
+  ORDINAL_STRING = { 1 => 'Primary', 2 => 'Secondary' }.freeze
+  scope :expired, -> { where('end_at < ?', current_time) }
+  scope :unexpired, -> { where('end_at >= ?', current_time) }
+  scope :active, -> { where('start_at < ? AND end_at >= ?', current_time, current_time) }
+  scope :primary, -> { where(ordinal: 1) }
+  scope :secondary, -> { where(ordinal: 2) }
+
+  validates :ordinal, presence: true, inclusion: { in: [1, 2] }
```

Comment:
we have constraints on the database, so adding checking in the app would be redundant.

Reply:
ah ok, i either missed the non-null db constraints yesterday, or those got added after i'd loaded the page (since i see GH saying "changed since last view").  apologies, agreed the db constraints are sufficient.